src/AspNet.Security.OAuth.SuperOffice/SuperOfficeAuthenticationPostConfigureOptions.cs

```diff
@@ -0,0 +1,79 @@
+        public void PostConfigure(
+            [NotNull] string name,
+            [NotNull] SuperOfficeAuthenticationOptions options)
+        {
+            if (string.IsNullOrEmpty(options.TokenValidationParameters.ValidAudience) && !string.IsNullOrEmpty(options.ClientId))
+            {
+                options.TokenValidationParameters.ValidAudience = options.ClientId;
+                options.TokenValidationParameters.ValidIssuer = options.ClaimsIssuer;
+            }
+
+            // As seen in:
+            // github.com/dotnet/aspnetcore/blob/master/src/Security/Authentication/OpenIdConnect/src/OpenIdConnectPostConfigureOptions.cs#L71-L102
+
+            // need this now to successfully instantiate ConfigurationManager below.
+            if (options.Backchannel == null)
+            {
+#pragma warning disable CA2000 // Dispose objects before losing scope
+                options.Backchannel = new HttpClient(options.BackchannelHttpHandler ?? new HttpClientHandler());
+#pragma warning restore CA2000 // Dispose objects before losing scope
+                options.Backchannel.DefaultRequestHeaders.UserAgent.ParseAdd("SuperOffice OpenIdConnect handler");
+                options.Backchannel.Timeout = options.BackchannelTimeout;
+                options.Backchannel.MaxResponseContentBufferSize = 1024 * 1024 * 10; // 10 MB
+            }
+
+            if (options.ConfigurationManager == null)
+            {
+                if (!(string.IsNullOrEmpty(options.MetadataAddress) && string.IsNullOrEmpty(options.Authority)))
+                {
+                    if (string.IsNullOrEmpty(options.MetadataAddress) && !string.IsNullOrEmpty(options.Authority))
+                    {
+                        options.MetadataAddress = options.Authority;
+                        if (!options.MetadataAddress.EndsWith("/", StringComparison.Ordinal))
+                        {
+                            options.MetadataAddress += "/";
+                        }
+
+                        options.MetadataAddress += ".well-known/openid-configuration";
+                    }
+
+                    if (options.RequireHttpsMetadata && !options.MetadataAddress.StartsWith("https://", StringComparison.OrdinalIgnoreCase))
+                    {
+                        throw new InvalidOperationException("The MetadataAddress or Authority must use HTTPS unless disabled for development by setting RequireHttpsMetadata=false.");
+                    }
+
+                    options.ConfigurationManager = new ConfigurationManager<OpenIdConnectConfiguration>(
+                        options.MetadataAddress,
+                        new OpenIdConnectConfigurationRetriever(),
+                        new HttpDocumentRetriever(options.Backchannel) { RequireHttps = options.RequireHttpsMetadata })
+                    {
+                        RefreshInterval = options.RefreshInterval,
+                        AutomaticRefreshInterval = options.AutomaticRefreshInterval,
```

Comment:
```suggestion
                        AutomaticRefreshInterval = options.AutomaticRefreshInterval,
                        RefreshInterval = options.RefreshInterval,
```

Reply:
I opted to set them without depending on the Option properties, which have since been removed.

Reply:
The comment here was purely related to sorting.

Reply:
I don't recall if it was this comment that motivated the change or something @kevinchalet mentioned.